ruby/ql/lib/codeql/ruby/regexp/RegExpTreeView.qll

```diff
@@ -268,6 +268,12 @@ class RegExpTerm extends RegExpParent {
 
   /** Gets the primary QL class for this term. */
   override string getAPrimaryQlClass() { result = "RegExpTerm" }
+
+  /** Holds if this regular expression term can match the empty string. */
+  predicate matchesEmptyString() { none() }
+
+  /** Gets a string matched by this regular expression. */
+  string getAMatch() { none() }
```

Comment:
In the JS library these methods are called `isNullable` and `getAMatchedString`. I actually like your names more, but perhaps maintaining consistency across languages is more important.